common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -157,6 +159,98 @@ class @Problem
 
   # TODO add hooks for problem types here by inspecting response.html and doing
   # stuff if a div w a class is found
+          if response.progress_status != "done"
+            @maybeSetupTimer()
+
+  getSecondsLeft: =>
+    # Returns 
+    #   a nonnegative integer representing seconds left on a timed exam.
+    #   -1 on error
+    if _.isUndefined(@startTime) or _.isUndefined(@totalSeconds)
+      return -1
+    else
+      return Math.floor(Math.max(@totalSeconds - ((new Date() - @startTime) / 1000), 0))
+
+  maybeSetupTimer: =>
+    if @el.find(".problem-timer").length
+      @$timer = @el.find('.problem-timer')
+      @$display = @el.find('.minutes-left')
+      @startTime = new Date()
+      @totalSeconds = parseInt(@$timer.data('totalSecondsLeft'), 10)
+      @secondsBeforeWarning = parseInt(@$timer.data('secondsBeforeWarning'), 10)
+      @submittedBeforeTimeExpired = false
+
+      # The timer is initially hidden by CSS 
+      @$timer.show()
+
+      # Clear old timers, eg. if we press the submit button
+      if @timerId
+        clearInterval(@timerId)
+
+      # Sync every second
+      @timerId = setInterval(@syncTimer, 1000)
+
+      # Initialize and show timer
+      @syncTimer()
```

Comment:
Ah I see. The template is "dynamic" and will be re-fetched from the server. The response header's Cache Control field is set to "no-cache, ..." so as to prevent client caching. 

I manually tested the case, but can set up a test case for this as well if needed.